internal/dwarf/frame/parser.go

```diff
@@ -0,0 +1,290 @@
+// Package frame contains data structures and
+// related functions for parsing and searching
+// through Dwarf .debug_frame data.
+package frame
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	"github.com/parca-dev/parca-agent/internal/dwarf/util"
+)
+
+// TODO(kakkoyun): Can we speed parsin using or look up using .eh_frame_hdr?
```

Comment:
Sorry just saw this comment! I don't think it's possible. We need to parse and evaluate all the entries. `.eh_frame_hdr`is designed for debuggers or tools that need to find the unwind information on-demand, rather than ahead-of-time, as we do in our case.

The reason for this is that they lazily parse and evaluate the instructions. Debuggers usually only need a few unwind tables to be fetched (and then they can cache them), generating the full tables would waste too many resources,

Reply:
Thanks for the explanation:heart: It's an old comment. I came to this realization recently as well.